data/lib/src/datasource_impl/shared_space_datasource_impl.dart

```diff
@@ -103,4 +103,21 @@ class SharedSpaceDataSourceImpl implements SharedSpaceDataSource {
       });
     });
   }
+
+  @override
+  Future<SharedSpaceNodeNested> createSharedSpaceWorkGroup(CreateWorkGroupRequest createWorkGroupRequest) {
+    return Future.sync(() async {
+      return (await _linShareHttpClient.createSharedSpaceWorkGroup(createWorkGroupRequest.toCreateWorkGroupBodyRequest())).toSharedSpaceNodeNested();
+    }).catchError((error) {
+      _remoteExceptionThrower.throwRemoteException(error, handler: (DioError error) {
+        if (error.response.statusCode == 404) {
+          throw SharedSpaceNotFound();
```

Comment:
Why does 404 is SharedSpaceNotFound ?

Reply:
That's the exception that occurs when this function fails. ` _linShareHttpClient.createSharedSpaceWorkGroup`

Reply:
ikr but ```SharedSpaceNotFound``` is when `sharedspaces/{id}` is not found